autoload/maktaba/path.vim

```diff
@@ -24,6 +20,39 @@ let s:drive_backslash = '\v^\a:\\\\'
 let s:drive_frontslash = '\v^\a://'
 
 
+""
+" Splits {path} on the last system separator character. Equivalent to using
+" @function(#Split) and re-joining all components but the last, but more
+" efficient since it's needed by performance critical code.
+function! s:SplitLast(path) abort
+  " First strip off root.
+  let l:root = maktaba#path#RootComponent(a:path)
```

Comment:
What's the point of stripping off the root here and reattaching it later? Seems like the linear search would work regardless.